### DISCUSSION TITLE
[Feat]: Support defining an offset value to the `scrollTo` method

### DISCUSSION THREAD

**Discussion by mririgoyen** [davidjerleke/embla-carousel#934] [Ideas]
### Which variants of Embla Carousel are relevant to this feature request?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Feature description

When using the `scrollTo` method to scroll to a specific index, it would be lovely if we could also provide an optional offset value so that the item isn't fully aligned to the edge of the container.

### Additional Context

For one of our implementations of the carousel (a carousel of tabs), the next and previous buttons sit on top of the tab bar. When we want to scroll programmatically to a specific tab, it aligns the tab all the way to the left edge, which means part of the tab is now covered by the previous button. Instead, it would be beneficial if we could scroll to a specific index and offset it by the width of the button, so it is not obscured.

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for feature requests

**Comment by mririgoyen** [comment]
Just as a follow up, I was able to achieve my desired behavior using the internal engine, although I would like to avoid that in the future, if possible.

```ts
const { scrollTarget, scrollTo } = api.internalEngine();
const { distance } = scrollTarget.byIndex(tabIndex, 1);
const offset = tabIndex > 0 ? 38 : 0;
scrollTo.distance(distance + offset, false);
```

**Reply by davidjerleke** [reply]
Hi @mririgoyen,
Nice going solving it yourself and thanks for sharing your solution. This may benefit other developers.

I understand that it would be convenient with an offset parameter for your use case, but I have to carefully consider any feature request that comes up in order to decide if the feature in question should be added or not. Up until now, this is the first time this feature request has come up so I need more traction before adding this feature to the core library. Of course, this doesn't have to mean that someone else haven't thought about it, but at least we can say that no one has wanted it enough to create a feature request.

I'm closing this for now and if it gets more traction in the future, I will consider adding it to the library core. I hope you understand.

Best,
David
